modules/system/classes/extensions/ModuleManager.php

```diff
@@ -0,0 +1,283 @@
+        $this->refresh($code);
+
+        // Return an instance of the module
+        return $this->get($code);
+    }
+
+    public function enable(WinterExtension|string $extension, string|bool $flag = self::DISABLED_BY_USER): mixed
+    {
+        // TODO: Implement enable() method.
+        throw new ApplicationException('Support for enabling modules needs implementing');
+    }
+
+    public function disable(WinterExtension|string $extension, string|bool $flag = self::DISABLED_BY_USER): mixed
+    {
+        // TODO: Implement disable() method.
+        throw new ApplicationException('Support for disabling modules needs implementing');
+    }
+
+    /**
+     * @throws ApplicationException
+     */
+    public function update(WinterExtension|string|null $extension = null, bool $migrationsOnly = false): ?bool
+    {
+        $modules = $this->getModuleList($extension);
+
+        $firstUp = UpdateManager::instance()->isSystemSetup();
+
+        if ($firstUp) {
+            $this->repository->createRepository();
+            $this->output->info('Migration table created');
+        }
+
+        if (!$migrationsOnly && !Config::get('cms.disableCoreUpdates')) {
+            foreach ($modules as $module) {
+                $extension = $this->get($module);
+                if (
+                    ($composerPackage = Composer::getPackageNameByExtension($extension))
+                    && Composer::updateAvailable($composerPackage)
+                ) {
+                    $this->output->info(sprintf(
+                        'Performing composer update for %s (%s) module...',
+                        $module,
+                        $composerPackage
+                    ));
+
+                    Preserver::instance()->store($extension);
+                    $update = Composer::update(dryRun: true, package: $composerPackage);
+
+                    $versions = $update->getUpgraded()[$composerPackage] ?? null;
+
+                    $versions
+                        ? $this->renderComponent(Info::class, sprintf('Updated module %s (%s) from v%s => v%s', $module, $composerPackage, $versions[0], $versions[1]))
+                        : $this->renderComponent(Error::class, sprintf('Failed to module %s (%s)', $module, $composerPackage));
+                } elseif (false /* Detect if market */) {
+                    Preserver::instance()->store($extension);
+                    // @TODO: Update files from market
+                }
+            }
+        }
+
+        foreach ($modules as $module) {
+            $this->output->info(sprintf('Migrating %s module...', $module));
+            $this->migrator->run(base_path() . '/modules/' . strtolower($module) . '/database/migrations');
+
+            if ($firstUp) {
+                $className = '\\' . $module . '\Database\Seeds\DatabaseSeeder';
+                if (class_exists($className)) {
+                    $this->output->info(sprintf('Seeding %s module...', $module));
```

Comment:
Probably out of scope for this PR, but should we approach seeders (both module and plugin, which aren't quite the same) as needing the same level of tracking as migrations so that any that haven't been run yet will get run? With the current code, if a new module is added after the project is setup then it will never have its seeder run because it'll never be the firstUp again.

Reply:
Agreed, maybe we could use `Parameter` to record inits for modules? It's currently used like that for themes.

Reply:
Works for me